src/Agent.Worker/Release/Artifacts/TfVCArtifact.cs

```diff
@@ -0,0 +1,150 @@
+        }
+
+        public IArtifactDetails GetArtifactDetails(IExecutionContext context, AgentArtifactDefinition agentArtifactDefinition)
+        {
+            var artifactDetails = JsonConvert.DeserializeObject<Dictionary<string, string>>(agentArtifactDefinition.Details);
+            var projectId = string.Empty;
+            var repositoryId = string.Empty;
+            var mappings = string.Empty;
+
+            if (!artifactDetails.TryGetValue(ArtifactDefinitionConstants.MappingsId, out mappings) || mappings == null)
+            {
+                string baseRepoPath = string.Join("/", "$", projectId);
+                var defaultMapping = new List<Dictionary<string, InputValue>>()
+                                         {
+                                             new Dictionary<string, InputValue>()
+                                                 {
+                                                     {
+                                                         ArtifactDefinitionConstants.ServerPathId,
+                                                         new InputValue { Value = baseRepoPath }
+                                                     },
+                                                     {
+                                                         ArtifactDefinitionConstants.MappingTypeId,
+                                                         new InputValue { Value = Constants.Release.Map }
+                                                     },
+                                                     {
+                                                         ArtifactDefinitionConstants.LocalPathId,
+                                                         new InputValue { Value = string.Empty }
+                                                     }
+                                                 }
+                                         };
+
+                mappings = JsonConvert.SerializeObject(defaultMapping);
+            }
+
+            if (artifactDetails.TryGetValue(ArtifactDefinitionConstants.ProjectId, out projectId)
+                && artifactDetails.TryGetValue(ArtifactDefinitionConstants.RepositoryId, out repositoryId))
+            {
+                return new TfVCArtifactDetails
+                {
+                    RelativePath = "\\",
+                    ProjectId = projectId,
+                    RepositoryId = repositoryId,
+                    Mappings = mappings
+                };
+            }
+            else
+            {
+                throw new InvalidOperationException(StringUtil.Loc("RMArtifactDetailsIncomplete", agentArtifactDefinition.Name));
+            }
+        }
+
+        private void PrepareTfVCEndpoint(ServiceEndpoint endpoint, TfVCArtifactDetails tfVcArtifactDetails)
+        {
+            var allMappings = JsonConvert.DeserializeObject<IList<Dictionary<string, InputValue>>>(tfVcArtifactDetails.Mappings);
```

Comment:
we don't support mapping for TfvcArtifact. should we do this when we start supporting?